src/OpenTelemetry.AutoInstrumentation.StartupHook/NativeMethods.cs

```diff
@@ -0,0 +1,49 @@
+// <copyright file="NativeMethods.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System.Runtime.InteropServices;
+
+namespace OpenTelemetry.AutoInstrumentation;
+
+internal static class NativeMethods
+{
+    private static readonly bool IsWindows = string.Equals(FrameworkDescription.Instance.OSPlatform, "Windows", StringComparison.OrdinalIgnoreCase);
+
+    public static bool IsProfilerAttached()
+    {
+        if (IsWindows)
+        {
+            return Windows.IsProfilerAttached();
+        }
+
+        return NonWindows.IsProfilerAttached();
+    }
+
+    // the "dll" extension is required on .NET Framework
+    // and optional on .NET Core
```

Comment:
This comment makes me think that we don't need Windows and NonWindows here since this is for .NET only.

Reply:
With that the dependency on FrameworkDescription can be removed.

Reply:
https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/pull/2897/commits/3b37535b0f92cedb9f16bb3a237436e81370646c

Reply:
Nice.